tests/Schema/AbstractTypeRegistryTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace SilverStripe\GraphQL\Tests\Schema;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\GraphQL\Schema\Storage\AbstractTypeRegistry;
+use SilverStripe\GraphQL\Controller as GraphQLController;
+use Symfony\Component\Filesystem\Filesystem;
+use ReflectionObject;
+use stdClass;
+
+class AbstractTypeRegistryTest extends SapphireTest
+{
+    /**
+     * Putting a .graphql-generated folder in a unit-test folder is similar to what IntegrationTest.php does
+     * It's important that we don't overwrite the real application /.graphql-generated folder
+     */
+    public const SOURCE_DIRECTORY = __DIR__ . '/.graphql-generated';
+
+    protected function setUp(): void
+    {
+        $dir = self::SOURCE_DIRECTORY;
+        if (!file_exists($dir)) {
+            mkdir($dir);
+        }
+    }
+
+    protected function tearDown(): void
+    {
+        $dir = self::SOURCE_DIRECTORY;
+        if (file_exists($dir)) {
+            $fs = new Filesystem();
+            $fs->remove($dir);
+        }
+        // ensure that any GraphqlController added to controller_stack is removed
+        (new Class extends Controller
+        {
+            public function removeGraphqlController()
+            {
+                if (((static::$controller_stack[0] ?? new stdClass()) instanceof GraphQLController)) {
+                    array_shift(static::$controller_stack);
+                }
+            }
+        })->removeGraphqlController();
+    }
+
+    public function testRebuildOnMissingPath()
+    {
+        list($registry, , $method) = $this->getInstance();
+        $this->assertSame(
+            self::SOURCE_DIRECTORY . '/' . AbstractTypeRegistry::REBUILD_ON_MISSING_FILENAME,
+            $method->invoke($registry)
+        );
+    }
+
+    /**
+     * @dataProvider provideRebuildOnMissing
+     */
+    public function testRebuildOnMissing(
+        string $message,
+        bool $controller,
+        bool $autobuild,
+        bool $config,
+        bool $interval,
+        bool $expected
+    ): void {
+        list($registry, $method) = $this->getInstance();
```

Comment:
```suggestion
        list($registry, $canRebuildOnMissing) = $this->getInstance();
```
Make it clear what the method is

Reply:
Updated